Production/scripts/validation_tool.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+import argparse
+parser = argparse.ArgumentParser('''
+The script runs a binned KS test between different chunks of the same RDataFrame. For simplicity, all chunks are compared to the first one.
+If a KS test is below the threshold, a warning message is printed on screen.
+NOTE: the binning of each variable must be hard coded in the script (using the BINS dictionary)
+NOTE: pvalue = 99 means that one of the two histograms is empty.
+''')
+
+import ROOT
+import glob
+import json
+from collections import OrderedDict
+
+parser.add_argument('--input'       , required = True, type = str, help = 'input file. Accepts glob patterns')
+parser.add_argument('--output'      , required = True, type = str, help = 'output file name')
+parser.add_argument('--pdf'         , default  = None, type = str, help = 'output pdf directory')
+parser.add_argument('--json'        , required = True, type = str, help = 'output json file name')
+parser.add_argument('--nsplit'      , default  = 5   , type = str, help = 'number of chunks per file')
+parser.add_argument('--pvthreshold' , default  = .05 , type = str, help = 'threshold of KS test (above = ok)')
+
+parser.add_argument('--visual', action = 'store_true', help = 'Won\'t run the script in batch mode')
+parser.add_argument('--legend', action = 'store_true', help = 'Draw a TLegent on canvases')
```

Comment:
add argument to specify number of threads and then set implicit multi-threading for ROOT:
```python
if args.n_threads > 1:
    ROOT.ROOT.EnableImplicitMT(args.n_threads)
```

Reply:
Unfortunatelly, "Range was called with ImplicitMT enabled. Multi-thread ranges are not supported."
Also, see https://root-forum.cern.ch/t/rdataframe-range-does-not-work-with-multi-thread/40466/2. I don't know if still holds for our root release
I don't know if there is any workaround

Reply:
full traceback:
```
Traceback (most recent call last):
  File "/cvmfs/cms.cern.ch/slc7_amd64_gcc700/external/python/2.7.14-pafccj/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/cvmfs/cms.cern.ch/slc7_amd64_gcc700/external/python/2.7.14-pafccj/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/cvmfs/cms.cern.ch/slc7_amd64_gcc700/external/python/2.7.14-pafccj/lib/python2.7/profile.py", line 610, in <module>
    main()
  File "/cvmfs/cms.cern.ch/slc7_amd64_gcc700/external/python/2.7.14-pafccj/lib/python2.7/profile.py", line 603, in main
    runctx(code, globs, None, options.outfile, options.sort)
  File "/cvmfs/cms.cern.ch/slc7_amd64_gcc700/external/python/2.7.14-pafccj/lib/python2.7/profile.py", line 77, in runctx
    prof = prof.runctx(statement, globals, locals)
  File "/cvmfs/cms.cern.ch/slc7_amd64_gcc700/external/python/2.7.14-pafccj/lib/python2.7/profile.py", line 444, in runctx
    exec cmd in globals, locals
  File "validation_tool.py", line 136, in <module>
    run_validation(dataframe = dataframe, pwd = main_dir, branches = ['lepton_gen_match', 'sampleType', 'dataset_group_id', 'dataset_id'])
  File "validation_tool.py", line 111, in run_validation
    histos = get_histos(dataframe, branch = branch, norm = True)
  File "validation_tool.py", line 61, in get_histos
    subframes = [dataframe.Range(ii*sub_size, (ii+1)*sub_size) for ii in range(N_SPLITS)]
TypeError: none of the 2 overloaded methods succeeded. Full details:
  ROOT::RDF::RInterface<ROOT::Detail::RDF::RRange<ROOT::Detail::RDF::RLoopManager>,void> ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void>::Range(unsigned int begin, unsigned int end, unsigned int stride = 1) =>
    Range was called with ImplicitMT enabled. Multi-thread ranges are not supported. (C++ exception of type runtime_error)
  ROOT::RDF::RInterface<ROOT::Detail::RDF::RRange<ROOT::Detail::RDF::RLoopManager>,void> ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void>::Range(unsigned int end) =>
    takes at most 1 arguments (2 given)
```

Reply:
I see. We planned that starting from the next steps (after the ShuffleMerge) we'll use LCG environment instead of the CMSSW-one. Probably it can be already the case for this tool - like this it could be run within the same environment as all further scripts.
Please, try to source LCG environment (as suggested in README.md) and try to run the script there. 
```sh
source /cvmfs/sft.cern.ch/lcg/views/LCG_97apython3/x86_64-centos7-clang10-opt/setup.sh
```

Reply:
looks like it is still unsupported

Reply:
ok... no implicit multi-threading then...